devicetypes/smartthings/aeon-multisensor-6.src/aeon-multisensor-6.groovy

```diff
@@ -398,46 +398,96 @@ def ping() {
+			break
+		case "0371-0002-0018":  // MultiSensor 7 EU
+		case "0371-0102-0018":  // MultiSensor 7 US
+		case "0371-0202-0018":  // MultiSensor 7 AU
+            		//1. set association groups for hub - 2 groups are used to set battery refresh interval different than sensor report interval
+            		request << zwave.associationV1.associationSet(groupingIdentifier: 1, nodeId: zwaveHubNodeId)
+
+            		//2. automatic report flags
+            		// param 101 -103 [4 bytes] 128: light sensor, 64 humidity, 32 temperature sensor, 15 ultraviolet sensor, 1 battery sensor
+            		// set value  241 (default for 101) to get all reports. Set value 0 for no reports (default for 102-103)
+            		//association group 1
+            		request << zwave.configurationV1.configurationSet(parameterNumber: 101, size: 1, scaledConfigurationValue: 240)
+
+            		//association group 2
+            		request << zwave.configurationV1.configurationSet(parameterNumber: 102, size: 1, scaledConfigurationValue: 1)
+
+            		//3. no-motion report x seconds after motion stops (default 30 secs)
+            		request << zwave.configurationV1.configurationSet(parameterNumber: 3, size: 2, scaledConfigurationValue: timeOptionValueMap[motionDelayTime] ?: 30)
+
+            		//4. motionSensitivity 3 levels: 6-normal, 11-maximum (default), 1-minimum, 0 - disabled
+            		request << zwave.configurationV1.configurationSet(parameterNumber: 4, size: 1,
+                		configurationValue:
+                    			motionSensitivity == "normal" ? [6] :
+                        			motionSensitivity == "minimum" ? [1] :
+                            				motionSensitivity == "disabled" ? [0] : [11])
+
+            		//5. Parameters 111-113: report interval for association group 1-3
+            		//association group 1 - set in preferences, default 8 mins
+            		request << zwave.configurationV1.configurationSet(parameterNumber: 111, size: 2, scaledConfigurationValue: timeOptionValueMap[reportInterval] ?: (8 * 60))
+
+            		//association group 2 - report battery every 6 hours
+            		request << zwave.configurationV1.configurationSet(parameterNumber: 112, size: 2, scaledConfigurationValue: 6 * 60 * 60)
+
+            		//6. report automatically ONLY on threshold change
+            		//From manual:
+            		//Enable/disable the selective reporting only when measurements reach a certain threshold or percentage set in 41-44.
+            		//This is used to reduce network traffic.  (0 = disable, 1 = enable)
+            		//Note: If USB power, the Sensor will check the threshold every 10 seconds. If battery power, the Sensor will check the threshold
+            		//when it is waken up.
+            		request << zwave.configurationV1.configurationSet(parameterNumber: 40, size: 1, scaledConfigurationValue: 1)
+			break
+	}
```

Comment:
a lot of these parameters are the same across models, couldn't you build a `request` list that has all the parameters in common and only separate out those parameters that are different inside the switch statement?

Reply:
Got it, i moved a bunch of the same lines from the switch statement outside if they are the same between MS6 and MS7. I left Parameter 3 since they have different range supports.